src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTryFinally.cs

```diff
@@ -2442,6 +2442,219 @@ .maxstack  2
 ");
         }
 
+        [WorkItem(18678, "https://github.com/dotnet/roslyn/issues/18678")]
+        [Fact]
+        public void TryCatchConstantFalseFilter1()
+        {
+            var src = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        try
+        {
+            throw new Exception();
+        }
+        catch (Exception) when (false)
```

Comment:
Please add a test for some expression that is statically known to be false, but not literally `false`, like `true == false`.